consensus/reactor.go

```diff
@@ -794,6 +793,38 @@ OUTER_LOOP:
 	}
 }
 
+func (conR *ConsensusReactor) peerStatsRoutine() {
+	for {
+		// Manage disconnects from self or peer.
+		if !conR.IsRunning() {
+			conR.Logger.Info("Stopping peerStatsRoutine")
+			return
+		}
+
+		select {
+		case msg := <-conR.conS.statsMsgQueue:
+			// Get peer
+			peer := conR.Switch.Peers().Get(msg.PeerID)
+			if peer != nil {
+				// Get peer state
+				ps := peer.Get(types.PeerStateKey).(*PeerState)
+				switch msg.MsgType {
+				case Vote:
+					if numVotes := ps.RecordVote(); numVotes%votesToContributeToBecomeGoodPeer == 0 {
```

Comment:
And also because we have a sense of overall peer contribution. This info could potentially be used break ties between two peers in case we need to choose. If we reset every 10k we lose this information.